lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Librarian.java

```diff
@@ -0,0 +1,24 @@
+package com.codedifferently.lesson10.library;
+
+public class Librarian extends Patron {
+
+  public Librarian(String name, String email) {
+    super(name, email);
+  }
+
+  public void addBook(Library library, Book book) {
```

Comment:
This isn't the right place to implement the requirement. It's still possible to add a book without a librarian by using the Library object directly. You need at least to modify the add/remove methods on the `Library` type.